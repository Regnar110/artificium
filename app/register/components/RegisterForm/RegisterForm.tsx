'use client'
import React from 'react'
import Button from '@mui/material/Button'
import Checkbox from '@mui/material/Checkbox';
import TextField from '@mui/material/TextField'
import { createTheme, ThemeProvider } from '@mui/material/styles';

import { useForm } from "react-hook-form";
const RegisterForm = () => {

    const { register, handleSubmit, watch, formState: { errors } } = useForm();
    const onSubmit = async data => {
        delete data.register_password_repeat
        const response = await fetch("http://localhost:3001/register",
            {
                method: "POST", // *GET, POST, PUT, DELETE, etc.
                mode: "cors", // no-cors, *cors, same-origin
                cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
                credentials: "same-origin", // include, *same-origin, omit
                headers: {
                "Content-Type": "application/json",
                // 'Content-Type': 'application/x-www-form-urlencoded',
                },
                redirect: "follow", // manual, *follow, error
                referrerPolicy: "no-referrer", // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
                body: JSON.stringify(data), // body data type must match "Content-Type" header
            }
        )
        const parsedResponse = await response.json()
        console.log(parsedResponse)
    }


    const theme = createTheme({
        palette: {
          primary: {
            main: '#82DBF7',
          }
        }
      })

  return (
    <form onSubmit={handleSubmit(onSubmit)}  className='register_form flex flex-col md:grid md:grid-cols-2 gap-5 h-fit md:row-span-2 w-full'>
        <ThemeProvider theme={theme}>
            <div className='register_input flex flex-col gap-y-4'>
                <span className='text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium'>E-mail</span>
                <TextField 
                    {
                        ...register(
                            "email",
                            {
                                required:"This field is required",
                                pattern: {value:/^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/, message:"Wrong email format"},
                            }
                        )
                    }
                    error={errors.mail?.message ? true : false}
                    helperText={errors.mail?.message ? errors.mail.message as string:""}
                    type='email'
                    InputLabelProps={{ style: {color:"#9B9C9E"}}} 
                    sx={{input: {color:"#fff", background:"#1A1D21", border:"2px solid #363A3D", borderRadius:"10px"}}} 
                    color="primary" 
                    id="outlined-basic" 
                    label="E-mail" 
                    variant="outlined" 
                />                        
            </div>
            <div className='register_input flex flex-col gap-y-4'>
                <span className='text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium'>Nickname</span>
                <TextField 
                    type='text'
                    {...register(
                        "nickname",
                        {
                            required:"This field is required",
                            minLength: { value:2, message:"This nickname is too short"},
                            maxLength:{ value:20, message:"This nickname is too long"}
                        }
                        )
                    }
                    error={errors.nickname?.message ? true:false}
                    helperText={errors.nickname?.message ? errors.nickname.message as string:""}
                    InputLabelProps={{ style: {color:"#9B9C9E"}}} 
                    sx={{input: {color:"#fff", background:"#1A1D21", border:"2px solid #363A3D", borderRadius:"10px"}}} 
                    color="primary" 
                    id="outlined-basic" 
                    label="Nickname" 
                    variant="outlined" 
                />
            </div>
            <div className='register_input flex flex-col gap-y-4'>
                <span className='text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium'>Password</span>
                <TextField 
                    type='password'
                    {...register(
                        "register_password",
                        {
                            required: "This field is required",
                            pattern: {value:/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/, message:"Wrong password format. At least 8 charaters, one big, one small, number and special character"},
                        }
                        )
                    }
                    error={errors.register_password?.message ? true:false}
                    helperText={errors.register_password?.message ? errors.register_password.message as string:""}
                    InputLabelProps={{ style: {color:"#9B9C9E"}}}
                    sx={{input: {color:"#fff", background:"#1A1D21", border:"2px solid #363A3D", borderRadius:"10px"}}} 
                    color="primary" 
                    id="outlined-basic" 
                    label="Password" 
                    variant="outlined" 
                />
            </div>
            <div className='register_input flex flex-col gap-y-4'>
                <span className='text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium'>Repeat password</span>
                <TextField 
                    type='password'
                    {...register(
                        "register_password_repeat",
                        {
                            required:"This field is required",
                            validate: (val:string) => {
                                if(watch("register_password") !== val) {
                                    return "Password not match"
                                }
                            }
                        }
                        )
                    }
                    error={errors.register_password_repeat?.message ? true : false} 
                    helperText={`${errors.register_password_repeat?.message ? errors.register_password_repeat.message:""}`} 
                    InputLabelProps={{ style: {color:"#9B9C9E"}}} 
                    sx={{input: {color:"#fff", background:"#1A1D21", border:"2px solid #363A3D", borderRadius:"10px"}}} 
                    color="primary" 
                    id="outlined-basic" 
                    label="Repeat password" 
                    variant="outlined" 
                />                            
            </div>
        </ThemeProvider>
        <div className='terms_conditions_checkbox flex gap-x-2 text-white font-plus_jakarta_sans items-center col-span-2'>
            <Checkbox 
                {...register(
                    "register_terms",
                    {
                        required:"You need to accept terms and agreements"
                    }
                    )
                }
                required={true}
                className='w-fit' 
                size='medium' 
                aria-label='terms_conditions' 
                sx={{color:"#363A3D", '&.Mui-checked': {color:"#B6F09C"}}} 
                defaultChecked={false}
            />
            <span className='text-[16px] w-fit'>I agree with <span className='font-plus_jakarta_sans text-[16px] font-bold text-transparent bg-clip-text gradient_dayblue_blue_green500'>Terms and conditions</span></span>
        </div>
        <Button
        type='submit'
        className='w-full col-span-2 bg-[#B6F09C] '
        sx={{
            backgroundColor:"#B6F09C",
            fontFamily:"Plus Jakarta Sans",
            fontWeight:"700",
            color:"#0C1132",
            ':hover': {
            bgcolor: '#9bf074', // theme.palette.primary.main
            },
        }}
        >
            Create free account
        </Button>
    </form>                
  )
}

export default RegisterForm
