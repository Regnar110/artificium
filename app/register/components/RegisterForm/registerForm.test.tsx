import { render, screen } from '@testing-library/react'
import RegisterForm from './RegisterForm'

import userEvent from '@testing-library/user-event'

const mock_FORM_SUBMIT_FN = jest.fn()

const renderTestedComponent = () => render(<RegisterForm/>)

beforeAll(()=> {
  userEvent.setup()
})

 
describe('RegisterForm component', () => {
  it('should render', () => {
    const {container:snapshotContainer} = renderTestedComponent()
    expect(snapshotContainer).toMatchSnapshot()
  })

  it('should render form fields', () => {
    renderTestedComponent()
    expect(screen.getByRole('register-form-element')).toBeTruthy();
    expect(screen.getByLabelText('E-mail')).toBeTruthy();
    expect(screen.getByLabelText('Nickname')).toBeTruthy();
    expect(screen.getByLabelText('Password')).toBeTruthy();
    expect(screen.getByLabelText('Repeat password')).toBeTruthy();
    expect(screen.getByRole('conditions_checkbox')).toBeTruthy();
    expect(screen.getByTestId('Register')).toBeTruthy();
  })

  it('submit button should call onSubmit method', async () => {
    renderTestedComponent();
    screen.getByRole('register-form-element').addEventListener('submit', mock_FORM_SUBMIT_FN)
    expect(userEvent.click(screen.getByTestId('Register'))).toHaveBeenCalled
  })

  // it('should register button call onSumbmit function', async () => {
  //   const {container} = render(<RegisterForm/>)
  //   console.log(container)
  //   const element = screen.getByRole('button')
  //   const response = await userEvent.click(element)
  //   console.log(response)
  // })
})