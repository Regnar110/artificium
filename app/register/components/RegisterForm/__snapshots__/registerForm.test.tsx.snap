// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component should render 1`] = `
<div>
  <form
    class="register_form flex flex-col md:grid md:grid-cols-2 gap-5 h-fit md:row-span-2 w-full"
    role="register-form-element"
  >
    <div
      class="register_input flex flex-col gap-y-4"
    >
      <span
        class="text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium"
      >
        E-mail
      </span>
      <div
        class="MuiFormControl-root MuiTextField-root css-skn84n-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-2tudcr-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic_email"
          id="outlined-basic_email-label"
          style="color: rgb(155, 156, 158);"
        >
          E-mail
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-d7oyt7-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic_email"
            name="email"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                E-mail
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="register_input flex flex-col gap-y-4"
    >
      <span
        class="text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium"
      >
        Nickname
      </span>
      <div
        class="MuiFormControl-root MuiTextField-root css-skn84n-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-2tudcr-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic_nickname"
          id="outlined-basic_nickname-label"
          style="color: rgb(155, 156, 158);"
        >
          Nickname
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-d7oyt7-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic_nickname"
            name="nickname"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Nickname
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="register_input flex flex-col gap-y-4"
    >
      <span
        class="text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium"
      >
        Password
      </span>
      <div
        class="MuiFormControl-root MuiTextField-root css-skn84n-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-2tudcr-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic_password"
          id="outlined-basic_password-label"
          style="color: rgb(155, 156, 158);"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-d7oyt7-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic_password"
            name="register_password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="register_input flex flex-col gap-y-4"
    >
      <span
        class="text-[14px] font-plus_jakarta_sans text-[#9B9C9E] font-medium"
      >
        Repeat password
      </span>
      <div
        class="MuiFormControl-root MuiTextField-root css-skn84n-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-2tudcr-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic_password_repeat"
          id="outlined-basic_password_repeat-label"
          style="color: rgb(155, 156, 158);"
        >
          Repeat password
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-d7oyt7-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic_password_repeat"
            name="register_password_repeat"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Repeat password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="text-[#d32f2f] font-plus_jakarta_sans"
    />
    <div
      class="terms_conditions_checkbox flex gap-x-2 text-white font-plus_jakarta_sans items-center col-span-2"
    >
      <span
        aria-label="terms_conditions"
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium w-fit css-1501mf5-MuiButtonBase-root-MuiCheckbox-root"
        role="conditions_checkbox"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          name="register_terms"
          required=""
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="text-[16px] w-fit"
      >
        I agree with 
        <span
          class="font-plus_jakarta_sans text-[16px] font-bold text-transparent bg-clip-text gradient_dayblue_blue_green500"
        >
          Terms and conditions
        </span>
      </span>
    </div>
    \\       
    <button
      class="w-full col-span-2 bg-[#B6F09C] min-h-[36.5px] flex justify-center items-center rounded-md"
      data-testid="Register"
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
